packages/cli/src/contracts/createx/deployCreate2Contract.ts

```diff
@@ -1,59 +1,59 @@
-import { Address, Hash, Hex, WalletClient } from "viem";
-import { estimateContractGas, simulateContract } from "viem/actions";
-import { CREATEX_ADDRESS, createXABI } from "./constants.js";
+import {Address, Hash, Hex, WalletClient} from 'viem';
+import {estimateContractGas, simulateContract} from 'viem/actions';
+import {CREATEX_ADDRESS, createXABI} from './constants.js';
 
 export type DeployCreate2ContractParameters = {
-    client: WalletClient
-    salt: Hex
-    initCode: Hex
-    createXAddress?: Address
-}
+	client: WalletClient;
+	salt: Hex;
+	initCode: Hex;
+	createXAddress?: Address;
+};
 
 export const deployCreate2Contract = async ({
-    client,
-    salt,
-    initCode,
-    createXAddress,
+	client,
+	salt,
+	initCode,
+	createXAddress,
 }: DeployCreate2ContractParameters): Promise<Hash> => {
-    const hash = await client.writeContract({
-        abi: createXABI,
-        chain: client.chain,
-        account: client.account?.address as Address,
-        address: createXAddress ?? CREATEX_ADDRESS,
-        functionName: 'deployCreate2',
-        args: [salt, initCode]
-    })
+	const hash = await client.writeContract({
+		abi: createXABI,
+		chain: client.chain,
+		account: client.account!,
+		address: createXAddress ?? CREATEX_ADDRESS,
+		functionName: 'deployCreate2',
+		args: [salt, initCode],
+	});
```

Comment:
this one where if u pass a privateKeyAccount it uses eth_sendrawtransaction but if just an address it uses eth_sendtransaction